### PR TITLE
Do not use `Loc.initial` as default parameter

### PR DESCRIPTION
**Pull request by thewilsonator**
This messes up generation of `frontend.h` referencing a member of Loc that doesn't exist (to the world of C++).

If this is deemed undesirable, the other way to fix this problem is to make `Loc.initial` `const __gshared` instead of `static immutable`.

### CODE REVIEW

File: compiler/src/dmd/mtype.d [hunk 1]
```diff
@@ -5851,9 +5851,9 @@ extern (C++) final class TypeEnum : Type
         return sym.getMemtype(loc).size(loc);
     }
 
-    Type memType(const ref Loc loc = Loc.initial)
+    Type memType()
```
Comment (thewilsonator) [inline compiler/src/dmd/mtype.d:5854]: this is never called with a non-default argument.

Comment (ibuclaw) [reply compiler/src/dmd/mtype.d:5854]: This is a `final` function, right? Couldn't you just make it `extern(D)`?

**Comment by Geod24** [comment]
Can we make the compiler error on those ? We should strive not to generate invalid headers...

**Comment by thewilsonator** [comment]
> Can we make the compiler error on those ? 

The convention seems to be to ignore parts of declarations that would yield invalid C++, so I did that.

File: compiler/src/dmd/mtype.d (hunk 1 above)
Comment (thewilsonator) [reply compiler/src/dmd/mtype.d:5854]: Correct. Maybe, I would consider the removal of an unused default argument a useful side effect refactoring.

**Pull request closed without merging**